### PR TITLE
Fix hang when exporting at the CLI

### PR DESCRIPTION
**Pull request by koppor**
Fixes https://github.com/JabRef/jabref/issues/10380

There were two issues:

- NPE when getting the main table preferences.
- Hanging after a successful export.

The former was solved by using the right getter ([`e4ea759` (#10383)](https://github.com/JabRef/jabref/pull/10383/commits/e4ea759377aa9f725cac439fef8fdc0edaff3f6d)), the latter was solved by shutting down the file update monitor [`18a3cd3` (#10383)](https://github.com/JabRef/jabref/pull/10383/commits/18a3cd3fe0c4b9101f039523d429304740dce7b1) (refs https://github.com/JabRef/jabref-issue-melting-pot/issues/290).

### Mandatory checks
- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/cli/ArgumentProcessor.java [hunk 1]
```diff
@@ -67,13 +67,20 @@ public class ArgumentProcessor {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(ArgumentProcessor.class);
     private final JabRefCLI cli;
-    private final List<ParserResult> parserResults;
+
+    // Written once by processArguments()
+    private List<ParserResult> parserResults;
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/cli/ArgumentProcessor.java:72]: init with empty list?

**Comment by Siedlerchr** [comment]
looks good so far, but would need testing with the binary.

**Comment by koppor** [comment]
> looks good so far, but would need testing with the binary.

Test was done at https://github.com/JabRef/jabref/issues/10380#issuecomment-1721333965.

I wonder whether we could simply do `System.exit(0);` instead doing a proper shut down after proper processing. -- I currently don't see any disadvantages there, but maybe, I overlook something?

**Comment by koppor** [comment]
I tried it with `System.exit(0);` at [`231b3a9` (#10383)](https://github.com/JabRef/jabref/pull/10383/commits/231b3a988adb84481f721fae4f4a34698c905255).

**Review by Siedlerchr — APPROVED** [review]

**Comment by Siedlerchr** [comment]
the 60s shutdown is normally only the timeout if the tasks were not shutdown before:
https://github.com/JabRef/jabref/blob/a0e2564806050247bce6056a6c108b87bab315b3/src/main/java/org/jabref/gui/Globals.java#L118-L130

**Pull request merged**